### PR TITLE
Feature/basic structure

### PR DESCRIPTION
**Pull request by ijosue16**
In this PR I added the following:

* #### Created class `Person` with  the following:
>   
>   * Instance vars: `@id`, `@name`, and `@age`.
>   * Constructor with `name`,  `age`, and `parent_permission` as parameters. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
>   * Getters for `@id`, `@name`, and `@age`.
>   * Setters for `@name` and `@age`.
>   * Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
>   * Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
 * ####  Created class `Student` with the following:
>   
>   * Inherits from `Person`.
>   * Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
>   * Method `play_hooky` that returns `"¯\(ツ)/¯"`.
 * ####  Created class `Teacher` with the following:
>   
>   * Inherits from `Person`.
>   * Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
>   * Override `can_use_services?` so it always returns `true`.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Person
+  attr_accessor :id, :name, :age
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @age = age
+    @name = name
```
Comment (m05e5) [inline person.rb]: - [x] Hi as part of the requirements, you are supposed to initialize the instance variable id too. so ill recommend initializing it with a unique random value
i.e `@id = Random.rand(1..1000)`

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || parent_permission
+  end
```
Comment (m05e5) [inline person.rb]: - [x] Great job setting the of_age method to private but, its only the of_age method that should be private not the can_use_services so ill recommend either putting the can_use_services before the `private` keyword or adding `public` before the `can_use_services`

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @ijosue16 , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [x] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Person
+  attr_accessor :id, :name, :age
+
+  def initialize(age, name = 'unknown', parent_permission: true)
```
Comment (christianbiring1) [inline person.rb:4]: [OPTIONAL]
To allow the user to put `parent_permission` while creating a person instance different than `true` if they want to, Kindly make sure 
```suggestion
  def initialize(age, name = 'unknown', parent_permission= true)
```

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(classroom, age, name = 'unknown', parent_permission: true)
```
Comment (christianbiring1) [inline student.rb:6]: [OPTIONAL]
To allow the user to put parent_permission while creating a person instance different than true if they want to, Kindly make sure
```suggestion
  def initialize(classroom, age, name = 'unknown', parent_permission= true)
```

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    @classroom = classroom
+    super(age, name, parent_permission: parent_permission)
```
Comment (christianbiring1) [inline student.rb:8]: [OPTIONAL]
- [x] Here parent_permission in the super method should then be
```suggestion
    super(age, name, parent_permission)
```

File: teacher.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+    @specialization = specialization
+    super(age, name, parent_permission: parent_permission)
+  end
```
Comment (christianbiring1) [inline teacher.rb:9]: [OPTIOANL]
- [x] As said early, to be able to add the variable `parent_permission` different than `true`
```suggestion
  def initialize(specialization, age, name = 'unknown', parent_permission = true)
    @specialization = specialization
    super(age, name, parent_permission)
  end
```

File: teacher.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+  def can_use_services?
+    'true'
```
Comment (christianbiring1) [inline teacher.rb:12]: [OPTIONAL]
- [x] Also here instead of returning the string `"true"` It would be great to return instead the boolean variable `true`

**Review by christianbiring1 — APPROVED** [review]
Hi @ijosue16 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 
Iapproveokay

## Highlights

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**